core/src/main/java/io/questdb/griffin/engine/functions/geohash/GeoHashNative.java

```diff
@@ -37,14 +43,63 @@ public static native void latestByAndFilterPrefix(
             long unIndexedNullCount,
             long maxValue,
             long minValue,
-            int partitionIndex,
+            int frameIndex,
             int blockValueCountMod,
-            long hashesAddress,
-            int hashLength,
+            int geoHashColumnIndex,
+            int geoHashColumnType,
             long prefixesAddress,
             long prefixesCount
-    );
+    ) {
+        long geoHashColumnAddress = 0;
+        // hashColumnIndex can be -1 for latest by part only (no prefixes to match)
+        if (geoHashColumnIndex > -1) {
+            final PageFrameMemory frameMemory = frameMemoryPool.navigateTo(frameIndex);
+            geoHashColumnAddress = frameMemory.getPageAddress(geoHashColumnIndex);
+        }
+
+        // -1 must be dead case here
+        final int geoHashColumnSize = ColumnType.isGeoHash(geoHashColumnType) ? getPow2SizeOfGeoHashType(geoHashColumnType) : -1;
```

Comment:
nit: maybe assert that `geoHashColumnSize != -1`?

Reply:
Good idea. Added the assertion in 5a553229e50cd1d14e4111a4b2e5d3a17de76b1d

Reply:
The previous patch was broken. The proper one is in 340e926c20032063f8e8fa15e28491997b83b239